File: Java/Intro/src/oop/Angajati.java
package oop;

public class Angajati {
    String nume;
    String prenume;
    String functie;
    double salariu;
    int vechime;

    // Constructor:
    public Angajati(String nume, String prenume, String functie, double salariu, int vechime) {
        this.nume = nume;
        this.prenume = prenume;
        this.functie = functie;
        this.salariu = salariu;
        this.vechime = vechime;
    }

    // Metode:
    public void descriereAngajat() {
        System.out.println("Numele: " + this.nume);
        System.out.println("Prenumele: " + this.prenume);
        System.out.println("Functia: " + this.functie);
        System.out.println("Vechime: " + this.vechime);
        System.out.println("Salariu: " + this.salariu);
        System.out.println();
    }

    // actualizare vechime angajat
    public void actualizareVechime() {
        System.out.println("Se actualizeaza vechimea pentru " + this.nume + " " + this.prenume +".");
        this.vechime += 1;
    }

    // marire salariu angajat in functie de vechime
    public void marireSalariu() {
        if (this.vechime <= 3) {
            this.salariu = this.salariu + 300;
        } else {
            this.salariu = this.salariu + 500;
        }
        System.out.println("S-a marit salariul pentru " + this.nume + " " + this.prenume);
    }
}

File: Python/Intro/oop/angajati.py
class Angajati:
    # constructorul:
    def __init__(self,nume,prenume,functie,vechime,salariu):
        self.nume = nume
        self.prenume = prenume
        self.functie = functie
        self.vechime = vechime
        self.salariu = salariu

    # metode:
    def descriereAngajat(self):
        print(f'Numele: {self.nume}')
        print(f'Prenumele: {self.prenume}')
        print(f'Functia: {self.functie}')
        print(f'Vechimea: {self.vechime}')
        print(f'Salariu: {self.salariu}')
        print()

    def actualizareVechime(self):
        print(f'Se actualizeaza vechimea pentru {self.nume} {self.prenume}.')
        self.vechime += 1
        print()

    def marireSalariu(self):
        if self.vechime >= 3:
            self.salariu = self.salariu + 500
        else:
            self.salariu = self.salariu + 300
        print(f'S-a marit salariul pentru {self.nume} {self.prenume}')
        print()